app/src/main/java/org/simple/clinic/summary/updatephone/UpdatePhoneNumberDialog.kt

```diff
@@ -0,0 +1,104 @@
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers.io
+import kotterknife.bindView
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.patient.Patient
+import org.simple.clinic.patient.PatientUuid
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.setTextAndCursor
+import javax.inject.Inject
+
+class UpdatePhoneNumberDialog : AppCompatDialogFragment() {
+
+  @Inject
+  lateinit var controller: UpdatePhoneNumberDialogController
+
+  private val phoneInputLayout by bindView<TextInputLayout>(R.id.updatephone_phone_inputlayout)
+  private val numberEditText by bindView<EditText>(R.id.updatephone_phone)
+  private val cancelButton by bindView<Button>(R.id.updatephone_cancel)
+  private val saveButton by bindView<Button>(R.id.updatephone_save)
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    TheActivity.component.inject(this)
+  }
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    return AlertDialog.Builder(requireContext())
+        .setTitle(R.string.patientsummary_updatephone_dialog_title)
+        .setMessage(R.string.patientsummary_updatephone_dialog_message)
+        .setView(R.layout.dialog_patientsummary_updatephone)
+        .create()
+  }
+
+  @SuppressLint("CheckResult")
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
+    super.onActivityCreated(savedInstanceState)
+
+    cancelButton.setOnClickListener {
+      dismiss()
+    }
+
+    Observable.merge(dialogCreates(), saveClicks())
+        .observeOn(io())
+        .compose(controller)
+  }
+
+  private fun dialogCreates(): Observable<UiEvent> {
+    val patientUuid = arguments!!.getSerializable(KEY_PATIENT_UUID) as PatientUuid
+    return Observable.just(UpdatePhoneNumberDialogCreated(patientUuid))
+  }
+
+  private fun saveClicks() =
+      RxView
+          .clicks(saveButton)
+          .map { UpdatePhoneNumberSaveClicked(number = numberEditText.text.toString()) }
+
+  fun showIncompletePhoneNumberError() {
+    TODO()
+  }
+
+  fun preFillPhoneNumber(number: String) {
+    numberEditText.setTextAndCursor(number)
+  }
+
+  companion object {
+    private const val TAG = "UpdatePhoneNumberDialog"
+    private const val KEY_PATIENT_UUID = "patientUuid"
+
+    fun show(patient: Patient, fragmentManager: FragmentManager) {
+      val existingFragment = fragmentManager.findFragmentByTag(TAG)
+
+      if (existingFragment != null) {
+        fragmentManager
+            .beginTransaction()
+            .remove(existingFragment)
+            .commitNowAllowingStateLoss()
+      }
+
+      val fragment = UpdatePhoneNumberDialog().apply {
+        arguments = Bundle(1).apply {
+          putSerializable(KEY_PATIENT_UUID, patient.uuid)
+        }
+      }
+
+      fragmentManager
+          .beginTransaction()
+          .add(fragment, TAG)
+          .commitNowAllowingStateLoss()
```

Comment:
Yep, it's just too easy to cause `IllegalStateException` crashes with fragments. If the Activity stops right when the dialog is to be shown, it'll result in a race condition and potentially crash. From the documentation of `commitNowAllowingStateLoss()`:

> This is dangerous because the commit can be lost if the activity needs to later be restored from its state

I'm okay with losing this commit if the Activity is stopping.

Reply:
What do you think?